pybop/_optimisation.py

```diff
@@ -180,10 +191,15 @@ def _run_pybop(self):
         final_cost : float
             The final cost associated with the best parameters.
         """
+        if "maxiter" in optimiser_kwargs.keys():
```

Comment:
Isn't this already defined in lines 99-105 above?

Reply:
Lines 99-105 apply properties defined by kwargs passed on initialisation, e.g.
`optim = pybop.Optimisation(cost, optimiser, maxiter=10)`
Whereas lines 194-197 apply further properties passed on a particular call, e.g.
`optim.run(maxiter=10)`
Passing kwargs via the run function will overwrite existing properties. Does this make sense?